src/components/Identity/createLegacyIdentity.js

```diff
@@ -55,7 +55,7 @@ export default ({ config, getEcidFromVisitor }) => {
       return Promise.resolve();
     },
     setEcid(ecid) {
-      if (idMigrationEnabled && !cookieJar.get(amcvCookieName)) {
+      if (idMigrationEnabled && getEcidFromLegacyCookies() !== ecid) {
```

Comment:
When the ECID changes we need to update the AMCV cookie. We don't want to overwrite the cookie otherwise because there could be other values within the cookie.